test/e2e/kubectl/kubectl.go

```diff
@@ -495,27 +496,27 @@ var _ = SIGDescribe("Kubectl client", func() {
 
 		ginkgo.It("should return command exit codes", func() {
```

Comment:
this will run all the test in this block in serial, maybe we can split it and run the commands that depend on pods restasrts in its own block so they run in parallel with the other tests?

Reply:
I'm having a hard time figuring this out.

The ginkgo docs say "`Its` can be randomized and parallelized across multiple nodes". 

So assuming I move these tests to their own `ginkgo.It` blocks that should be enough?

Testing locally I haven't been able to get these to run concurrently.

I'm probably missing a flag?

```
make WHAT=test/e2e/e2e.test

_output/bin/e2e.test --ginkgo.focus="eddie test" --provider=local --kubeconfig=/Users/ezaneski/.kube/config --ginkgo.parallel.total=30 --test.parallel=30
```

Reply:
Yeah, you need to run the Gingko binary to parallelize not the e2e binary ,
Check the script used in ci https://github.com/kubernetes/kubernetes/blob/1f87e3fe0069eccaa0b0c5885022290c84b7799c/hack/ginkgo-e2e.sh#L171

Reply:
Ahh I got it - thank you!

```
_output/bin/ginkgo --nodes 3 --focus="Simple pod" --skip="should support exec through kubectl proxy|should support exec through an HTTP proxy|should handle in-cluster config" _output/bin/e2e.test -- --provider=local --kubeconfig=/Users/ezaneski/.kube/config
```

Run time is down to 6.5 minutes now.

I pushed up my changes but not sure if that's the correct blocks to use. I'll play with it more when I wake up.

Reply:
this is a simplified version
```
# setting this env prevents ginkgo e2e from trying to run provider setup
export KUBERNETES_CONFORMANCE_TEST='y'
# setting these is required to make RuntimeClass tests work ... :/
export KUBE_CONTAINER_RUNTIME=remote
export KUBE_CONTAINER_RUNTIME_ENDPOINT=unix:///run/containerd/containerd.sock
export KUBE_CONTAINER_RUNTIME_NAME=containerd
# FIXME we should not tolerate flakes
# but until then, we retry the test in the same job
# to stop PR retriggers for totally broken code
export FLAKE_ATTEMPTS=5
export NUM_NODES=20
# Kind clusters are three node clusters
export NUM_WORKER_NODES=3
ginkgo --nodes=${NUM_NODES} \
	--focus=${FOCUS} \
	--skip=${SKIPPED_TESTS} \
	--flakeAttempts=${FLAKE_ATTEMPTS} \
	/usr/local/bin/e2e.test \
	-- \
	--kubeconfig=${HOME}/admin.conf \
	--provider=local \
	--dump-logs-on-failure=false \
	--report-dir=${E2E_REPORT_DIR}	\
	--disable-log-dump=true \
	--num-nodes=${NUM_WORKER_NODES}
```